agent/engine/task_manager.go

```diff
@@ -57,6 +58,9 @@ var (
 	_stoppedSentWaitInterval       = stoppedSentWaitInterval
 	_maxStoppedWaitTimes           = int(maxStoppedWaitTimes)
 	taskNotWaitForSteadyStateError = errors.New("managed task: steady state check context is nil")
+	// Based on https://github.com/aws/amazon-ecs-agent/blob/master/agent/vendor/github.com/docker/docker/client/request.go#L182
+	startContainerEOFErrorRegex = "error during connect: Post .+: EOF"
```

Comment:
should we catch all EOFs responses for the startContainer workflow and handle them similarly? vs matching on that  first part of the error string as well. wyt?

Reply:
yeah make sense. updated to check that error ends in EOF, instead of hardcoding the error as a regex